lib/doorkeeper/oauth/pre_authorization.rb

```diff
@@ -5,6 +5,7 @@ module OAuth
     class PreAuthorization
       include Validations
 
+      validate :attributes, error: :invalid_request
```

Comment:
Maybe `validate_params` ? Like HTTP request params. Just a "sugar"

Reply:
Indeed at that time, I followed this line
https://github.com/doorkeeper-gem/doorkeeper/blob/25c5bda05df1c03770b8c1e999858942eb8f0eb0/lib/doorkeeper/oauth/authorization_code_request.rb#L6

Reply:
Oh, we can replace it too :)